run-build-functions.sh

```diff
@@ -1015,28 +1015,27 @@ install_go() {
     fi
   fi
 
-  if [ "$GIMME_GO_VERSION" != "$installGoVersion" ]
+  # Cache known version for 7days (604800 seconds)
+  resolvedGoVersion="$(GIMME_KNOWN_CACHE_MAX=604800 gimme --resolve $installGoVersion)"
```

Comment:
I wonder if doing this now always can impact performance or even error rate. This needs network and then caches the results for 7 days.

If a real version is supplied like 1.19 or 1.19.3 this will not access network. Only if the `installGoVersion ` is `stable` or `1.19.x` which is now also the default.